File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationArriveEvent.java
package com.mapbox.services.android.navigation.v5.navigation.metrics;

import android.annotation.SuppressLint;

@SuppressLint("ParcelCreator")
class NavigationArriveEvent extends NavigationEvent {
  private static final String NAVIGATION_ARRIVE = "navigation.arrive";

  NavigationArriveEvent(PhoneState phoneState) {
    super(phoneState);
  }

  @Override
  protected String getEventName() {
    return NAVIGATION_ARRIVE;
  }
}

File: libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/metrics/NavigationArriveEvent.kt
package com.mapbox.services.android.navigation.v5.navigation.metrics

import android.annotation.SuppressLint

private val NAVIGATION_ARRIVE = "navigation.arrive"

@SuppressLint("ParcelCreator")
class NavigationArriveEvent @JvmOverloads constructor(phoneState: PhoneState, override val eventName: String = NAVIGATION_ARRIVE) : NavigationEvent(phoneState) {
    fun eventName() = NAVIGATION_ARRIVE

}
